hid.pyx

```diff
@@ -113,7 +135,9 @@ cdef class device:
         cdef char* cbuff = path
         self._c_hid = hid_open_path(cbuff)
         if self._c_hid == NULL:
-            raise IOError('open failed')
+            raise IOError("Failed to open device")
```

Comment:
Unfortunately the only way to differentiate between HIDAPI and other possible IO errors was to check for the `'open failed'` argument.

This difference is important to detect and report to the user a common "not enough privileges" condition.

Thus, I ask that this not be changed unless done in backwards compatible form.

---

Other IOErrors have been changed in this pull request, and could cause similar issues for other dependent projects.

Reply:
I see your point. I would suggest creating a few custom Exception classes and raise those instead. Catching specific exceptions is then also made easier for users. Do you agree?

Reply:
I agree that custom exceptions would be a major improvement.  Using the information from `hid_error` could also be beneficial, either when instantiating the exception or, what's probably better, lazily with some `details` method.

For backwards compatibility, the new exceptions could extend `OSError` (`IOError` is just an alias to it these days).  In most cases that would probably be enough, and changing the "message" (the first argument) would be fine.  But I would keep it in the open failed exception, because that's probably the one that's currently most handled.

_(And, due to the same packaging difficulties mentioned on the other comment, most depend projects will need to remain compatible with previous cython-hidapi versions)._

Reply:
Alright.
What do you mean with the last part? Can we change the message or do we need to keep the messages identical to what they are now?

Reply:
I think the 'open failed' messages are the only that are important to keep.  In the other cases it's already possible to differentiate between the major failure cases by the exception type (ValueError vs OSError).

That said, I'm not personally affected by changes to the messages of the other exceptions, so I'm biased.